src/algebra/order/complete_field.lean

```diff
@@ -312,3 +312,26 @@ instance : unique (β ≃+*o γ) := unique_of_subsingleton $ induced_order_ring_
 
 end induced_map
 end linear_ordered_field
+
+section real
+
+/-- Any ring homomorphism `ℝ → ℝ` is an ordered ring isomorphism, and thus there are no
+nontrivial ring homorphism `ℝ → ℝ`. -/
+lemma real.ring_hom_eq_id (f : ℝ →+* ℝ) : f = ring_hom.id ℝ :=
+begin
+  have f_mon : monotone f,
+  { intros a b hab,
+    have : f b - f a ≥ 0,
+    { calc
+        f b - f a = f (b - a)                 : (ring_hom.map_sub _ _ _).symm
+          ...     = f (real.sqrt (b - a)^2)   : by rw sq_sqrt (sub_nonneg.mpr hab)
+          ...     = (f (real.sqrt (b - a)))^2 : ring_hom.map_pow _ _ _
+          ...     ≥ 0                         : sq_nonneg _, },
+    linarith, },
+  let φ : ℝ →+*o ℝ := { to_ring_hom := f, monotone' := f_mon },
+  let ι : ℝ →+*o ℝ := { to_ring_hom := ring_hom.id ℝ, monotone' := monotone_id, },
+  haveI : subsingleton (ℝ →+*o ℝ):= order_ring_hom.subsingleton,
+  exact congr_arg (λ f : ℝ →+*o ℝ, f.to_ring_hom) (subsingleton.elim φ ι),
```

Comment:
```suggestion
  haveI : subsingleton (ℝ →+*o ℝ):= order_ring_hom.subsingleton,
  exact congr_arg order_ring_hom.to_ring_hom
    (subsingleton.elim ⟨f, f_mon⟩ ⟨ring_hom.id ℝ, monotone_id⟩),
```
Up to you (untested); I'll remove the `haveI` in #16046 if your PR is merged first.